BedrockServer.h

```diff
@@ -8,30 +8,95 @@
 class BedrockServer : public SQLiteServer {
   public:
 
-    // Shutting down a bedrock server correctly is a multi-step process that ensures we will still respond to any
-    // requests we received right up until we are about to shut down.
+    // Shutting Down and Standing Down a BedrockServer. 
+    //
+    // # What's the difference between these two things?
+    // 
+    // Shutting down is pretty obvious - when we want to turn a server off, we shut it down. The shut down process
+    // tries to do this without interrupting any client requests.
+    // Standing down is a little less obvious. If we're the master node in a cluster, there are two ways to stand down.
+    // The first is that we are shutting down, in which case we'll need to let the rest of the cluster know that it
+    // will need to pick a new master. The other is if a higher-priority master asks to stand up in our place. In this
+    // second case, we'll stand down without shutting down.
+    //
+    //
+    // # Shutting Down
+    // Let's start with shutting down. Standing down is a subset of shutting down (when we start out mastering), and
+    // we'll get to that later.
+    //
+    // When a BedroskServer comes up, it's _shutdownState is RUNNING. This is the normal operational state. When the
+    // server receives a SIGINT, that state changes to START_SHUTDOWN. This change causes a couple things to happen:
```

Comment:
this isn't strictly SIGINT -- there are other signals that cause us to shut down (i.e. we use SIGTERM)